### PR TITLE
Fixed one of the possible errors as part of SAIC-357

### PR DESCRIPTION
**Pull request by realsystem**

### CODE REVIEW

**Comment by roman-verchikov** [comment]
Please add unit tests to cover that case

File: cloudferrylib/utils/files.py [hunk 1]
```diff
@@ -68,7 +68,10 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.runner.run_ignoring_errors(_unlink_dir(self.dirname))
-        return self
+        if exc_type is None:
+            return self
+        else:
+            raise exc_type
```
Comment (roman-verchikov) [inline cloudferrylib/utils/files.py]: Would be better to simply delete `return` statement: it would re-raise exception and keep all the necessary exception information untouched, while in your solution you're hiding `exc_val` and `exc_tb`.

**Comment by realsystem** [comment]
-1

**Comment by realsystem** [comment]
ready for review, +1

File: cloudferrylib/utils/remote_runner.py [hunk 2]
```diff
@@ -53,10 +53,17 @@ def run(self, cmd):
...
-                else:
-                    return run(cmd)
+                res = None
+                #import pdb;pdb.set_trace()
+                try:
+                    if self.sudo and self.user != 'root':
+                        res = sudo(cmd)
+                    else:
+                        res = run(cmd)
+                except RemoteExecutionError as e:
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: Exception must not be handled here, it must be handled by calling code. Please remove.

File: cloudferrylib/utils/files.py [hunk 3]
```diff
@@ -68,7 +68,8 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.runner.run_ignoring_errors(_unlink_dir(self.dirname))
-        return self
+        if exc_type is not None:
+            raise exc_type, exc_val, exc_tb
```
Comment (roman-verchikov) [inline cloudferrylib/utils/files.py]: Simply removing `return self` would suffice here

File: cloudferrylib/utils/remote_runner.py [hunk 4]
```diff
@@ -53,10 +53,17 @@ def run(self, cmd):
             with forward_agent(self.key):
                 LOG.debug("running '%s' on '%s' host as user '%s'",
                           cmd, self.host, self.user)
-                if self.sudo and self.user != 'root':
-                    return sudo(cmd)
-                else:
-                    return run(cmd)
+                res = None
+                #import pdb;pdb.set_trace()
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: `import pdb`...

File: cloudferrylib/utils/remote_runner.py (hunk 2 above)
Comment (realsystem) [inline cloudferrylib/utils/remote_runner.py]: Why?

File: cloudferrylib/utils/remote_runner.py (hunk 4 above)
Comment (realsystem) [inline cloudferrylib/utils/remote_runner.py]: sure

**Comment by realsystem** [comment]
ready for review

File: cloudferrylib/utils/remote_runner.py [hunk 5]
```diff
@@ -53,10 +53,12 @@ def run(self, cmd):
...
                 LOG.debug("running '%s' on '%s' host as user '%s'",
                           cmd, self.host, self.user)
+                res = None
                 if self.sudo and self.user != 'root':
-                    return sudo(cmd)
+                    res = sudo(cmd)
                 else:
-                    return run(cmd)
+                    res = run(cmd)
+                return res
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: Please undo this change, it does not do anything.

File: cloudferrylib/utils/drivers/ssh_chunks.py [hunk 6]
```diff
@@ -45,7 +45,11 @@ def splitter(total_size, block_size):
 
 
 def remote_file_size(runner, path):
-    return int(runner.run('stat --printf="%s" {path}'.format(path=path)))
+    res = runner.run('stat --printf="%s" {path}'.format(path=path))
+    if res is not None:
+        return int(res)
+    else:
+        raise RuntimeError("Can not get information about file")
```
Comment (roman-verchikov) [inline cloudferrylib/utils/drivers/ssh_chunks.py]: This is not needed. If value returned by `stat` cannot be converted to `int` – it means there's something completely wrong here.

File: tests/cloudferrylib/utils/test_file_utils.py [hunk 7]
```diff
@@ -137,3 +135,15 @@ def test_retries_if_error_occurs(self, scp, _):
...
             assert scp.call_count == num_retries + 1
+    def my_run(self, cmd):
+        raise Exception("ttt")
+    def test_temp_dir_exception_inside_with(self):
+        runner = mock.Mock()
+        dirname = 'dir'
+
+        try:
+            with files.RemoteDir(runner, dirname):
+                self.my_run("ls rrr")
```
Comment (roman-verchikov) [inline tests/cloudferrylib/utils/test_file_utils.py]: let's just simply `raise` from here, there's no need in extra *public* method.

**Comment by roman-verchikov** [comment]
+1
